qobject/block-qdict.c

```diff
@@ -224,7 +224,7 @@ void qdict_array_split(QDict *src, QList **dst)
     for (i = 0; i < UINT_MAX; i++) {
         QObject *subqobj;
         bool is_subqdict;
-        QDict *subqdict = NULL;
```

Comment:
I guess you wanted to keep a clean set of changes over qemu, however I was unable to compile without this using gcc 12.1.1, as it complained explicitly that this might not be initialized. Probably this should be fixed in upstream, worth checking if the 7.0 branch of qemu has this issue.

Reply:
The upstream doesn't have it, I suppose it's because GCC 12.1.1 is relatively new (less than a month...2022-05-06).
I checked it compiles well with GCC 11.2. If you don't have any hard requirements on the version of GCC I propose to keep it like this, otherwise I'll revert this change?

Reply:
No hard requirements, I'm just using Fedora, which means I typically end up on bleeding edge stuff faster than other distros. I can just compile via docker anyway so no problems, let's keep it so we match upstream.